src/sagas/SignIn.ts

```diff
@@ -0,0 +1,43 @@
+import { Frost } from '@poetapp/frost-client'
+import { Actions } from 'actions/index'
+import { browserHistory } from 'react-router'
+import { delay, SagaIterator } from 'redux-saga'
+import { call, takeLatest, put, ForkEffect } from 'redux-saga/effects'
+const { toast } = require('react-toastify')
+
+export async function signInFrost(data: {
```

Comment:
Is this function just a wrapper around `frost.login` which hard-codes `{ host: '/api' }`?

If so, is it possible to just

```ts
const { token } = yield call(frost.login, email, password)
```

and avoid having to declare this wrapper?

Is there a pattern for injecting a singleton-like instance to sagas, so Frost is instantiated once and readily available to all sagas?

The examples in the redux-saga docs all seem to `import Api from '...'` — treating it as a singleton. We could do the same if we instantiated it at the root of the application, before we wire the sagas. But maybe there's a better approach?

Reply:
I haven't used sagas very many times, so I was using frost-web as an example for how to implement these. I went this route because that is how it was handled there. 

I'm trying to just instantiate  where the sagas are called so that I can do 

```js
const frost = new Frost({ host: '/api' })
export function* SignIn(action: any): SagaIterator {
  try {
    const { email, password } = action.payload
    yield put(Actions.LoadingPage.onLoadingOn())
    const { token } = yield call(frost.login, email, password)
```

I think I am misunderstanding something though. Having trouble getting this to work.

Reply:
Alright, so I just gave this a shot and it seems to be working:

Check it out: https://github.com/poetapp/explorer-web/pull/134/files

I think it reduces code complexity a bit by removing an entire function and still achieving the same result. 

Let me know what you think.